src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.ValidateSignature.cs

```diff
@@ -245,10 +245,9 @@ private static ValidationResult<SecurityKey> ValidateSignatureWithKey(
                     new MessageDetail(
                         TokenLogMessages.IDX10518,
                         result.UnwrapError().MessageDetail.Message),
-                    ValidationFailureType.SignatureValidationFailed,
-                    typeof(SecurityTokenInvalidSignatureException),
-                    new StackFrame(true),
-                    result.UnwrapError());
+                    ValidationFailureType.SignatureAlgorithmValidationFailed,
```

Comment:
I am not comfortable with this change. We are modifying the behaviour of the method here by changing the exception type being returned.
The original path returns a `SecurityTokenInvalidSignatureException` with the `SecurityTokenInvalidAlgorithmException` as the `innerException`, and that is what this code was doing.

I believe we agreed that we should respect the existing exception types being returned and that we would only modify the error messages being returned.

`ValidateJWSAsync` has a special behaviour for "recoverable exception types" which as of now are `SecurityTokenInvalidSignatureException`, `SecurityTokenInvalidIssuerException`, and `SecurityTokenSignatureKeyNotFoundException`.

Could you please elaborate on the reason for this change?

Reply:
This is the comment, @jennyf19.

We discussed it, and accepted the explanation that the intention here is to return more specific errors, and not consider the invalid algorithm in the signature as a recoverable exception/error.

So now the case where the signature is done with an unsupported algorithm will return a `SecurityTokenInvalidAlgorithmException` instead of the original `SecurityTokenInvalidSignatureException` wrapping a `SecurityTokenInvalidAlgorithmException`, which was considered recoverable.